File: entrance_module/src/main/java/com/example/entrance_module/EntranceActivity.java
package com.example.entrance_module;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.alibaba.android.arouter.launcher.ARouter;

public class EntranceActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_entrance);
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ARouter.getInstance().build("/main/HomeActivity").navigation();
            }
        });
    }
}

File: entrance_module/src/main/java/com/example/entrance_module/EntranceActivity.kt
package com.example.entrance_module

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.alibaba.android.arouter.launcher.ARouter
import com.example.common_module.utils.RouteUtils
import kotlinx.android.synthetic.main.activity_entrance.*

class EntranceActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_entrance)
        button.setOnClickListener {
           RouteUtils.routerActivity(RouteUtils.MAIN_ACTIVITY)

        }
    }
}
